### PR TITLE
add `optionalAsync` API reference

### PR DESCRIPTION
**Pull request by EltonLobo07**

### CODE REVIEW

File: website/src/routes/api/(async)/optionalAsync/index.mdx [hunk 1]
```diff
@@ -1,10 +1,235 @@
...
+);
+```
+
+### New user schema
+
+Schema to validate new user details.
+
+```ts
+const NewUserSchema = v.objectAsync({
+  // Assume the email is checked
```
Comment (fabian-hiller) [inline website/src/routes/api/(async)/optionalAsync/index.mdx:72]: What do you mean by that comment? It is not entirely clear to me.

**Comment by fabian-hiller** [comment]
Rest looks good to me.

File: website/src/routes/api/(async)/optionalAsync/index.mdx (hunk 1 above)
Comment (EltonLobo07) [reply website/src/routes/api/(async)/optionalAsync/index.mdx:72]: I meant the uniqueness of the email is already checked. This assumption helps to completely skip email async validation like in the example shown below.
```ts
const NewUserSchema = v.objectAsync({
  email: v.pipeAsync(
    v.string(),
    v.email(),
    v.checkAsync(isEmailUnique, 'The email is not unique')
  ),
  username: v.optionalAsync(
    v.pipeAsync(
      v.string(),
      v.nonEmpty(),
      v.checkAsync(isUsernameUnique, 'The username is not unique.')
    )
  ),
  password: v.pipe(v.string(), v.minLength(8)),
});
```
The problem with the above example is that it is sending a wrong message: 'perform backend validation for every field in the object'. I would have restructured my example to this:
```ts
const NewUserSchema = v.pipeAsync(
  v.objectAsync({
    email: v.pipe(v.string(), v.email()),
    username: v.optional(v.pipe(v.string(), v.nonEmpty())),
    password: v.pipe(v.string(), v.minLength(8)),
  }),
  v.partialCheckAsync(
    [['email'], ['username']],
    areUserDetailsUnique,
    'The user details are not unique.'
  )
);
```
But the above example won't demonstrate how to use `optionalAsync`

Do you think I should change the example? If not, what would be your code comment to make the intention clear?

Comment (fabian-hiller) [reply website/src/routes/api/(async)/optionalAsync/index.mdx:72]: I would use this code example:

```ts
const NewUserSchema = v.objectAsync({
  email: v.pipeAsync(
    v.string(),
    v.email(),
    v.checkAsync(isEmailUnique, 'The email is not unique')
  ),
  username: v.optionalAsync(
    v.pipeAsync(
      v.string(),
      v.nonEmpty(),
      v.checkAsync(isUsernameUnique, 'The username is not unique.')
    )
  ),
  password: v.pipe(v.string(), v.minLength(8)),
});
```

**Pull request merged**